pkg/visitors/find.go

```diff
@@ -157,6 +165,58 @@ func FindAndPredicate(predicate1 FindPredicate, predicate2 FindPredicate) FindPr
 	}
 }
 
+// FindExactlyOne does a find using a supplied predicate and errors if there's more than one.
+func FindExactlyOne(f uefi.Firmware, pred func(f uefi.Firmware) bool) (uefi.Firmware, error) {
+	find := &Find{
+		Predicate: pred,
+	}
+	if err := find.Run(f); err != nil {
+		return nil, err
+	}
+	// There should only be one match, there should only be one Dxe Core.
+	if mlen := len(find.Matches); mlen != 1 {
+		return nil, fmt.Errorf("expected exactly one match, got %v, matches were: %v", mlen, find.Matches)
+	}
+	return find.Matches[0], nil
+}
+
+// FindEnclosingFV finds the FV that contains a file.
+func FindEnclosingFV(f uefi.Firmware, file *uefi.File) (*uefi.FirmwareVolume, error) {
+	pred := func(f uefi.Firmware) bool {
+		switch f := f.(type) {
+		case *uefi.FirmwareVolume:
+			for _, v := range f.Files {
+				if v == file {
+					return true
+				}
+			}
+		}
+		return false
+	}
+	dxeFV, err := FindExactlyOne(f, pred)
+	if err != nil {
+		return nil, fmt.Errorf("unable to find DXE FV, got: %v", err)
+	}
+	// result must be a FV.
+	fv := dxeFV.(*uefi.FirmwareVolume)
```

Comment:
could you do this in a way that won't panic on a type assertion? You could do a type switch:
switch v := dxeFV.(type) {
case *uefi.FirmwareVolume:
default:
fmt.Errorf("%v: got %v, want *uefi.FirmwareVolume", dxeFV, v)
}

I suspect this will be better in the long run.

Reply:
I actually had a check in there, but I felt it wasn't necessary, since we have a predicate that can only ever match FVs. I figured if it ever changed enough that we suddenly weren't getting that back a panic was warranted.

Reply:
Done